sdks/go/pkg/beam/staticcheck.conf

```diff
@@ -0,0 +1,21 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO(BEAM-14371): Clean up instances of S1021 (merge variable declaration and assignment) and enable check
+# TODO(BEAM-14372): Clean up instances of U1000 (unused fields, vars, functions) and enable check
+checks = ["all", "-ST1000", "-ST1003", "-ST1016", "-ST1020", "-ST1021", "-ST1022", "-ST1023", "-SA1019", "-U1000", "-S1021"]
+initialisms = ["ACL", "API", "ASCII", "CPU", "CSS", "DNS", "EOF", "GUID", "HTML", "HTTP", "HTTPS", "ID", "IP", "JSON", "QPS", "RAM", "RPC", "SLA", "SMTP", "SQL", "SSH", "TCP", "TLS", "TTL", "UDP", "UI", "GID", "UID", "UUID", "URI", "URL", "UTF8", "VM", "XML", "XMPP", "XSRF", "XSS", "SIP", "RTP", "AMQP", "DB", "TS"]
+dot_import_whitelist = ["github.com/mmcloughlin/avo/build", "github.com/mmcloughlin/avo/operand", "github.com/mmcloughlin/avo/reg"]
```

Comment:
what does this configuration do?

Reply:
This is the actual configuration file that staticcheck uses. This is the default config (https://staticcheck.io/docs/configuration/#example-configuration) with the addition of skipping three other checks. In the event that we cleaned up the extra warnings from things like deprecations we could remove this, but as we've discussed the deprecation warnings are a decently sized project and getting some signal from this sooner is better

Reply:
Default configuration that `staticcheck` uses doesn't mean it's appropriate for our project though.

Technically, I think we should simply remove the
`initialisms`, `dot_import_whitelist`, and `http_status_code_whitelist` since we aren't using anything more than the defaults anyway, and it removes the implication we're relying on these specific configurations.
We certainly should keep the `checks` as we've previously discussed.

We don't (and shouldn't) have dot imports in the SDK, and we don't use these packages ourselves, so we shouldn't allow this gap to have them introduced.

Reply:
Fair enough, removing those extra fields doesn't impact our results from the tool anyway